aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -476,175 +490,288 @@ module AptosFramework::Token {
+        royalty_points_nominator: u64,
+        mutate_setting: vector<bool>,
+        property_keys: vector<String>,
+        property_values: vector<vector<u8>>,
+        property_types: vector<String>
+    ): TokenId acquires Collections, TokenStore {
+        /*
+            TODO: capability and signer check
+       */
+        let token_mut_config = create_token_mutability_config(&mutate_setting);
+
+        let tokendata_id = create_tokendata(
+            account,
+            collection,
+            name,
+            description,
+            0, // haven't minted token yet
+            maximum,
+            uri,
+            royalty_payee_address,
+            royalty_points_denominator,
+            royalty_points_nominator,
+            token_mut_config,
+            property_keys,
+            property_values,
+            property_types
         );
-        Table::add(
-            &mut collections.mint_capabilities,
-            token_id,
-            MintCapability { token_id },
+
+        let token_id = mint_token(
+            account,
+            tokendata_id,
+            balance,
+            Vector::empty<String>(),
+            Vector::empty<vector<u8>>(),
+            Vector::empty<String>()
         );
 
-        if (initial_balance > 0) {
-            initialize_token_store(account);
-            initialize_token(account, token_id);
-            mint(account, Signer::address_of(account), token_id, initial_balance);
+        token_id
+    }
+
+    fun get_properties_to_be_updated(
+        token_data: &TokenData,
+        keys: &vector<String>,
+        values: &vector<vector<u8>>,
+        types: &vector<String>
+    ): PropertyList {
+        let res_keys = Vector::empty<String>();
+        let res_vals = Vector::empty<vector<u8>>();
+        let res_types = Vector::empty<String>();
+        let default_properties = &token_data.default_property_list;
+        let i = 0;
+        while (i < Vector::length(keys)) {
+            let k = Vector::borrow(keys, i);
+            let v = Vector::borrow(values, i);
+            let t = Vector::borrow(types, i);
+            if (PropertyList::contains(default_properties, k) ) {
+                if ( PropertyList::get_type(default_properties, k) == *t && PropertyList::get_value(default_properties, k) != *v ) {
+                    Vector::push_back(&mut res_keys, *k);
+                    Vector::push_back(&mut res_vals, *v);
+                    Vector::push_back(&mut res_types, *t);
+                };
+            };
+            i = i + 1;
         };
+        PropertyList::new(&res_keys, &res_vals, &res_types)
+    }
+
+    public fun mint_token(
+        account: &signer,
+        token_data_id: TokenDataId,
+        amount: u64,
+        keys: vector<String>,
+        values: vector<vector<u8>>,
+        types: vector<String>
+    ): TokenId acquires Collections, TokenStore {
+        /*
+        TODO: capability and signer check
+        */
```

Comment:
let's focus on signer for today and capability after we agree

Reply:
we may not have signer check and only rely on capability. We can add this in the following PR for TokenV1